File: src/backend/Services/Reservations/OrangeCarRental.Reservations.Application/DTOs/ReservationDto.cs
namespace SmartSolutionsLab.OrangeCarRental.Reservations.Application.DTOs;

/// <summary>
///     Data transfer object for reservation details.
/// </summary>
public sealed record ReservationDto(
    Guid ReservationId,
    Guid VehicleId,
    Guid CustomerId,
    DateOnly PickupDate,
    DateOnly ReturnDate,
    string PickupLocationCode,
    string DropoffLocationCode,
    int RentalDays,
    decimal TotalPriceNet,
    decimal TotalPriceVat,
    decimal TotalPriceGross,
    string Currency,
    string Status,
    string? CancellationReason,
    DateTime CreatedAt,
    DateTime? ConfirmedAt,
    DateTime? CancelledAt,
    DateTime? CompletedAt);

File: src/frontend/libs/reservation-api/src/lib/models/reservation.dto.ts
/**
 * Reservation data model
 */
import type { Currency, ISODateString, Price } from "@orange-car-rental/shared";
import type { VehicleId } from "@orange-car-rental/vehicle-api";
import type { LocationCode } from "@orange-car-rental/location-api";
import type { ReservationId } from "./reservation-id.type";
import type { CustomerId } from "./customer-id.type";
import type { ReservationStatus } from "./reservation-status.enum";

export type Reservation = {
  readonly id: ReservationId;
  readonly vehicleId: VehicleId;
  readonly customerId: CustomerId;
  readonly pickupDate: ISODateString;
  readonly returnDate: ISODateString;
  readonly pickupLocationCode: LocationCode;
  readonly dropoffLocationCode: LocationCode;
  readonly rentalDays: number;
  readonly totalPriceNet: Price;
  readonly totalPriceVat: Price;
  readonly totalPriceGross: Price;
  readonly currency: Currency;
  readonly status: ReservationStatus;
  readonly cancellationReason: string | null;
  readonly createdAt: ISODateString;
  readonly confirmedAt: ISODateString | null;
  readonly cancelledAt: ISODateString | null;
  readonly completedAt: ISODateString | null;
};
